legit.py

```diff
@@ -0,0 +1,53 @@
+from git import Repo
+from git.exc import GitCommandError, InvalidGitRepositoryError
+from typing import Iterable
+
+# parent_commits: Tuple[str, str] = (), author: Tuple[str, str] = author,
+# committer: Tuple[str, str] = committer
+# TODO Consider author and commiter param
+# debug and rewrite clone method and others
+# author = Actor("An author", "katridi@yandex.ru")
+# committer = Actor("A committer", "katridi@yandex.ru")
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
+            Repo(path)
+        except InvalidGitRepositoryError:
+            Repo.clone_from(url, path)
+        return cls(path)
+
+    def create_branch(self, branch_name: str) -> None:
+        # Check if branch exist in other case create new branch
+        try:
+            self.repo.git.checkout('-b', branch_name)
+        except GitCommandError:
+            self.repo.git.checkout(branch_name)
+
+    def commit(self, meaningful_msg: str) -> None:
+        self.repo.index.commit(meaningful_msg)
+
+    def add_changes(self, files: Iterable[str] = ()) -> None:
+        [self.repo.index.add(file) for file in files] or self.repo.git.add(A=True)
```

Comment:
cannot understand what this method does

Reply:
Original intention was to get iterator with paths and add them throught list comprehensions or use 'git add .' analog if iterable variable is empty

Reply:
Ok this is a case where alternative execution is justified because it matches user experience in CLI. What I still have a problem with is that the falsy-ness is used not even for the parameter but for some derivative of a parameter. The list is further not used anywhere, creating doubt that a list really needs to be formed. A much simpler code and more readable code would do: 
```suggestion
        if files: 
            for file in files:
                self.repo.index.add(file) 
        else: 
            self.repo.git.add(A=True)
```

Reply:
Also, there is no need to initialise files with an empty tuple in this case, a None would do.

Reply:
Yes, you right. None is a good choice